transcendence-app/src/user/user.service.ts

```diff
@@ -14,6 +14,20 @@ export class UserService {
   retrieveUserWithId(id: string) {
     return this.users.find((user) => user.id === id) || null;
   }
+  retrieveUsers() {
+    return this.users;
+  }
+  retrieveUsernames(take: number, skip: number, sort: boolean): string[] {
+    const ret: string[] = [];
+    if (take < 0 || skip < 0 || skip >= this.users.length) {
+      throw new BadRequestException('Provided values incorrect');
+    }
+    take = skip + take > this.users.length ? this.users.length : skip + take;
+    for (let i = skip; i < take; i++) {
+      ret.push(this.users[i].username);
+    }
```

Comment:
Lets do it the good js way :yum:

```suggestion
    ret = this.users
      .filter((user, i) => i >= skip && i < take)
      .map(user => user.name);
```

Reply:
Good js solution. I'll try something like that